.github/workflows/build-push-image.yml

```diff
@@ -0,0 +1,48 @@
+name: Create and publish a Docker image
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Docker metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+              ghcr.io/pinecone-io/canopy
+          tags: |
```

Comment:
Don't we want to add a `:latest` tag pointing to the latest stable version, and a `:nightly` tag pointing at latest push?

Reply:
Latest will be automatically generated (https://github.com/docker/metadata-action?tab=readme-ov-file#latest-tag) Basically it will not be called nightly but every merge to main will deploy with docker tag main (override). We can also move to nightly with cron jobs docker metadata action has support for it but I don't think our PR pace justifies that as of now.